### PR TITLE
[IMP] Create pack operation lots in a single query

### PR DESCRIPTION
**Pull request by StefanRijnhart**
Replacing ORM creation by SQL brings this stanza back from 16 minutes to 3 seconds to create some 100.000 records.

Also, take care that the correct quantities are set. Earlier version uses `qty_done`, which is only setup a couple of lines later (and even then is only setup for operations of done pickings): https://github.com/OCA/OpenUpgrade/blob/9.0/addons/stock/migrations/9.0.1.1/post-migration.py#L59. All quantities came out zero that way.

### CODE REVIEW

**Comment by StefanRijnhart** [comment]
cc @paulius-sladkevicius

**Comment by paulius-sladkevicius** [comment]
@StefanRijnhart speed, speed, love it! LGTM

File: addons/stock/migrations/9.0.1.1/post-migration.py [hunk 1]
```diff
@@ -35,21 +35,23 @@ def _migrate_pack_operation(env):
...
+        env.cr,
+        """ INSERT INTO stock_pack_operation_lot (
+            id, lot_id, operation_id, qty_todo, qty,
+            create_uid, write_uid, create_date, write_date)
+        SELECT
+            nextval('stock_pack_operation_lot_id_seq'),
+            %(lot_id)s, o.id,
+            CASE WHEN p.state != 'done' THEN o.product_qty ELSE 0 END,
+            CASE WHEN p.state = 'done' THEN o.product_qty ELSE 0 END,
+            1, 1, now() at time zone 'UTC', now() at time zone 'UTC'
```
Comment (pedrobaeza) [inline addons/stock/migrations/9.0.1.1/post-migration.py]: Better to use here the same user as the stock_pack_operation, and even the dates (although this one is more debatable)

Comment (StefanRijnhart) [reply addons/stock/migrations/9.0.1.1/post-migration.py]: Done!

**Review by pedrobaeza — APPROVED** [review]

**Pull request merged**